src/annotator/test/config-test.js

```diff
@@ -1,86 +1,220 @@
-    fakeScriptConfig = '{"annotations":"456"}';
-    assert.deepEqual(config(fakeWindowBase), {
-      app: 'app.html',
-      annotations: '456',
+  context("when there's a window.hypothesisConfig() function", function() {
+    it('reads arbitrary settings from hypothesisConfig() into options', function() {
+      var window_ = fakeWindow();
+      window_.hypothesisConfig = sinon.stub().returns({foo: 'bar'});
+
+      var options = config(window_);
+
+      assert.equal(options.foo, 'bar');
     });
+
+    specify('hypothesisConfig() settings override js-hypothesis-config ones', function() {
+      var window_ = fakeWindow();
+      window_.hypothesisConfig = sinon.stub().returns({
+        foo: 'fooFromHypothesisConfigFunc'});
+      fakeSettings.returns({foo: 'fooFromJSHypothesisConfigObj'});
+
+      var options = config(window_);
+
+      assert.equal(options.foo, 'fooFromHypothesisConfigFunc');
+    });
+
+    context('if hypothesisConfig() returns a non-object value', function() {
+      it("doesn't add anything into the options", function() {
+        var window_ = fakeWindow();
+        window_.hypothesisConfig = sinon.stub().returns(42);
+
+        var options = config(window_);
+
+        delete options.app; // We don't care about options.app for this test.
+        assert.deepEqual({}, options);
+      });
+    });
+  });
+
+  context("when window.hypothesisConfig() isn't a function", function() {
+    it('throws a TypeError', function() {
+      var window_ = fakeWindow();
+      window_.hypothesisConfig = 'notAFunction';
+
+      assert.throws(
+        function() { config(window_); }, TypeError,
+        'hypothesisConfig must be a function, see: https://h.readthedocs.io/en/latest/embedding.html'
+      );
+    });
+  });
+
+  describe('showHighlights', function() {
+    context("when it's true", function() {
+      it('changes it to "always"', function () {
+        fakeSettings.returns({showHighlights: true});
+
+        var options = config(fakeWindow());
+
+        assert.equal(options.showHighlights, 'always');
+      });
+    });
+
+    context("when it's false", function() {
+      it('changes it to "never"', function () {
+        fakeSettings.returns({showHighlights: false});
+
+        var options = config(fakeWindow());
+
+        assert.equal(options.showHighlights, 'never');
+      });
+    });
+
+    context("when it's a string", function() {
+      it('passes it through unmodified', function () {
+        // It adds any arbitrary string value for showHighlights to the
+        // returned options, unmodified.
+        fakeSettings.returns({showHighlights: 'foobar'});
+
+        var options = config(fakeWindow());
+
+        assert.equal(options.showHighlights, 'foobar');
+      });
+    });
+  });
+
+  it("extracts the annotation query from the parent page's URL", function() {
+    config(fakeWindow());
+
+    assert.calledOnce(fakeExtractAnnotationQuery.extractAnnotationQuery);
```

Comment:
Given that the `extract-annotation-query` module exports one function, it would be more consistent with the rest of our code if it just returned that function as the default export (ie. `module.exports = <function name>`) instead of exporting it as a named export.